source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -651,6 +659,16 @@ namespace AdaptiveCards { namespace Uwp
                 XamlHelpers::AppendXamlElementToPanel(newControl.Get(), parentPanel);
                 childCreatedCallback(newControl.Get());
             }
+            else
+            {
+                HString elementTypeWrapper;
+                elementTypeWrapper.Set(elementType);
+                std::wstring errorString = L"No Renderer found for type: ";
+                errorString += elementTypeWrapper.GetRawBuffer(nullptr);
+                renderContext->AddFailure(
+                    FailureStatusCode::NoRendererForType,
+                    HStringReference(errorString.c_str()).Get());
```

Comment:
Any suggestions on how to make this better?

Reply:
Not really, but elementType should be an HString to begin with, then you don't need the extra wrapper variable.

---
In reply to: [151178751](https://github.com/Microsoft/AdaptiveCards/pull/950#discussion_r151178751) [](ancestors = 151178751)

Reply:
Yep, good call.

---
In reply to: [151264474](https://github.com/Microsoft/AdaptiveCards/pull/950#discussion_r151264474) [](ancestors = 151264474,151178751)